draft-oid4vc-haip-sd-jwt-vc.md

```diff
@@ -186,37 +186,120 @@ In addition, this profile defines the following additional requirements.
+
+The following is a non-normative example of an object comprising `credentials_supported` parameter of Credential format `vc+sd-jwt`.
+
+<{{examples/credential_metadata_sd_jwt_vc.json}}
+
+### Credential Offer
+
+The following additional claims are defined for this Credential format.
+
+* `credential_definition`: REQUIRED. JSON object containing the detailed description of the credential type. It MUST contain at least `type` property as defined in (#server_metadata_vc_sd-jwt).
+
+The following is a non-normative example of an object comprising `credentials_supported` parameter of Credential format `vc+sd-jwt`.
+
+<{{examples/credential_offer_sd_jwt_vc.json}}
+
+### Authorization Details {#authorization_vc_sd-jwt}
+
+The following additional claims are defined for authorization details of type `openid_credential` and this Credential format.
+
+* `credential_definition`: REQUIRED.  JSON object containing the detailed description of the credential type. It MUST contain at least `type` property as defined in (#server_metadata_vc_sd-jwt). It MAY contain `claims` property as defined in (#server_metadata_vc_sd-jwt).
+
+The following is a non-normative example of an authorization details object with Credential format `vc+sd-jwt`.
+
+<{{examples/authorization_details_sd_jwt_vc.json}}
+
+### Credential Request
+
+The following additional parameters are defined for Credential Requests and this Credential format.
+
+* `credential_definition`: REQUIRED. JSON object containing the detailed description of the credential type. It MUST contain at least `type` property as defined in (#server_metadata_vc_sd-jwt). It MAY contain `claims` property as defined in (#server_metadata_vc_sd-jwt).
+
+The following is a non-normative example of a Credential Request with Credential format `vc+sd-jwt`.
+
+<{{examples/credential_request_sd_jwt_vc.json}}
+
+### Credential Response {#credential_response_jwt_vc_json}
+
+The value of the `credential` claim in the Credential Response MUST be a JSON string that is an SD-JWT VC. Credentials of this format are already suitable for transfer and, therefore, they need not and MUST NOT be re-encoded.
+
+The following is a non-normative example of a Credential Response with Credential format `vc+sd-jwt`.
+
+<{{examples/credential_response_sd_jwt_vc.txt}}
+
+### Verifier Metadata
+
+The Verifier SHOULD add a `vp_formats_supported` element to its metadata (e.g. in the `client_metadata` authorization request parameter) to let the wallet know what protection algorithms it supports in conjunction with SD-JWT VCs. The format element MUST have the key `vc+sd-jwt`, the value is an object consisting of the following elements:
+
+* `sd-jwt-alg`: OPTIONAL. A JSON array containing identifiers of cryptographic algorithms the verifier supports for protection of a SD-JWT. If present, the `alg` JOSE header (as defined in [@!RFC7515]) of the presented SD-JWT MUST match one of the array values.
```

Comment:
sd-jwt-alg -> sd-jwt_signing_alg_values_supported
This uses the metadata naming conventions in https://openid.net/specs/openid-connect-discovery-1_0.html and related OAuth specs.

Reply:
verifier is client to it is not discovery spec, but client registration spec that is relevant, so `sd-jwt-alg` is correct.

Reply:
If you want examples from the client registration spec, the names are the likes of `id_token_signed_response_alg`, `userinfo_signed_response_alg`, `request_object_signing_alg`, `token_endpoint_auth_signing_alg`, etc.

The point of the requested changes is to have HAIP be clearly aligned with other related specs and the conventions used in them - and not appear to be something that was done in isolation, ignoring the way closely related specifications work.  Alignment will be good for everyone and for the acceptance of HAIP.

As a detailed point, whenever an element is a list of items, we use a plural name.  And when the list is of things supported, we use "supported" in the name.

Hence, I stand by my request to use names that will be unsurprising to people already familiar with OAuth and OpenID naming conventions:
`sd-jwt-alg` -> `sd-jwt_signing_alg_values_supported`
`kb-jwt-alg` -> `kb-jwt_signing_alg_values_supported`

Reply:
sure, sd_jwt_signing_alg might be more aligned - please open an issue or a PR to track this.

but no need for `_values_supported` because this is Client metadata and not AS metadata.

Reply:
The description of the field includes
> A JSON array containing identifier**s** of cryptographic algorithms the verifier **supports**

It's a list of supported values, so "supported" is called for.  And because it's a list, the name needs to be plural, not singular.

I'm ready to approve this PR after the changes are applied.  Given I was suggesting naming alignment, I didn't expect my review comments to be at all controversial.

Reply:
@selfissued these are the specs we are aligning to because verifier is the client,
https://openid.net/specs/openid-connect-registration-1_0.html
https://datatracker.ietf.org/doc/html/rfc7591
and the convention here is not to use `_supported` - please check.

I made changes to `*_signng_alg` - please re-review